### PR TITLE
Java 11 readiness: build both on JDK8 and JDK11

### PR DESCRIPTION
**Pull request by batmat**
Making sure this plugin is constantly checked both building with JDK8 and JDK11.

This goes with the Jenkins Java 11 General Availability announcement made back in March, and we're making a pass to check plugins are looking good on a JDK11.

([to understand the Jenkinsfile content](https://wiki.jenkins.io/display/JENKINS/Java+11+Developer+Guidelines#Java11DeveloperGuidelines-MakesureyourpluginistestedinContinuousIntegrationonJava8andJava11atthesametime))

@jenkinsci/java11-support

### CODE REVIEW

**Comment by batmat** [comment]
Thanks a lot for the help @bitwiseman !

File: src/main/java/jenkins/branch/WorkspaceLocatorImpl.java [hunk 1]
```diff
@@ -233,6 +233,14 @@ private static FilePath locate(TopLevelItem item, String fullName, Node node, bo
         return map;
     }
 
+    // Workaround spotbugs false-positive redunant null checking for try blocks in java 11
+    // https://github.com/spotbugs/spotbugs/issues/756
+    private static @NonNull InputStream readFilePath(@NonNull FilePath index) throws IOException, InterruptedException {
```
Comment (stephenc) [inline src/main/java/jenkins/branch/WorkspaceLocatorImpl.java]: We do not use Glick-style annotations. If the method returns `@NonNull` then the `@NonNull` shall be on the line preceding the access modifiers not between the access modifiers and the return type

**Comment by batmat** [comment]
@stephenc if that now suits your requests, could you please merge it? Thanks!

File: src/main/java/jenkins/branch/WorkspaceLocatorImpl.java [hunk 2]
```diff
@@ -233,6 +233,15 @@ private static Map<String, String> load(FilePath workspace) throws IOException,
         return map;
     }
 
+    // Workaround spotbugs false-positive redunant null checking for try blocks in java 11
```
Comment (jglick) [inline src/main/java/jenkins/branch/WorkspaceLocatorImpl.java:236]: #347

**Pull request merged**